File: C/Easy/1431.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
bool *kidsWithCandies(int *candies, int candiesSize, int extraCandies,
                      int *returnSize)
{
    int max = candies[0];
    // get the true max
    for (int i = 1; i < candiesSize; i++)
    {
        if (candies[i] > max)
        {
            max = candies[i];
        }
    }

    bool *answer = malloc(candiesSize * sizeof(bool));
    for (int i = 0; i < candiesSize; i++)
    {
        if (candies[i] + extraCandies >= max)
        {
            answer[i] = true;
        }
        else
        {
            answer[i] = false;
        }
    }
    *returnSize = candiesSize;
    return answer;
}

File: Java/Easy/1431.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        // find the greatest number of candy first
        int max = candies[0];
        for (int i : candies) {
            if (i > max) {
                max = i;
            }
        }

        // iterate through all
        List<Boolean> answer = new ArrayList<>();
        for (int i : candies) {
            if (i + extraCandies >= max) {
                answer.add(true);
            } else {
                answer.add(false);
            }
        }

        return answer;
    }
}